src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.Expressions.cs

```diff
@@ -862,24 +860,6 @@ private void EmitMemberGet(MemberInfo member, Type objectType)
             }
         }
 
-        private static bool TryGetRawConstantValue(FieldInfo fi, out object value)
-        {
-            // TODO: It looks like GetRawConstantValue is not available at the moment, use it when it comes back.
-            //value = fi.GetRawConstantValue();
-            //return true;
-
-            try
-            {
-                value = fi.GetValue(obj: null);
-                return true;
-            }
-            catch
-            {
-                value = null;
```

Comment:
`GetValue` and `GetRawConstantValue` are `virtual`/`abstract` members on `FieldInfo`, and `Attributes` (used to decide on `IsLiteral`) is also `abstract`, so there's indeed a possibility of an exception being thrown for custom subtypes of `FieldInfo`. The original code didn't protect against that; we wouldn't get very far anyway when emitting against such a custom reflection object.

@VSadov, what do you think? Keep the exception handling and put in a justification in a comment, or revert back to the .NET Framework code for `EmitMemberGet` that lacks such a handler?